tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,441 @@
+  const DataVector sin_phi = sin(phi);
+  const DataVector cos_theta = cos(theta);
+  const DataVector sin_theta = sin(theta);
+
+  test_surface_tangents[0].get(0) =
+      cos_phi * cos_theta * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[0].get(1) =
+      cos_theta * sin_phi * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[0].get(2) =
+      amp * square(cos_theta) * sin_phi -
+      sin_theta * (radius + amp * sin_phi * sin_theta);
+  test_surface_tangents[1].get(0) =
+      -radius * sin_phi + amp * sin_theta * (2. * square(cos_phi) - 1);
+  test_surface_tangents[1].get(1) =
+      cos_phi * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[1].get(2) = amp * cos_phi * cos_theta;
+
+  // Create DataBox
+  auto box =
+      db::create<db::AddTags<StrahlkorperDB::TagList<Frame::Inertial>::Tags>,
+                 db::AddComputeItemsTags<StrahlkorperDB::TagList<
+                     Frame::Inertial>::ComputeItemsTags>>(strahlkorper);
+
+  const auto& surface_tangents =
+      db::get<StrahlkorperDB::Tangents<Frame::Inertial>>(box);
+  for (size_t i = 0; i < 2; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      CHECK_ITERABLE_APPROX(gsl::at(surface_tangents, i).get(j),
+                            gsl::at(test_surface_tangents, i).get(j));
+    }
+  }
+
+  // Test surface_cartesian_coordinates
+  StrahlkorperDB::types<Frame::Inertial>::Vector test_cart_coords(n_pts);
+
+  {
+    const auto temp = radius + amp * sin_phi * sin_theta;
+    test_cart_coords.get(0) = cos_phi * sin_theta * temp + center[0];
+    test_cart_coords.get(1) = sin_phi * sin_theta * temp + center[1];
+    test_cart_coords.get(2) = cos_theta * temp + center[2];
+  }
+  const auto& cart_coords =
+      db::get<StrahlkorperDB::CartesianCoords<Frame::Inertial>>(box);
+  for (size_t j = 0; j < 3; ++j) {
+    CHECK_ITERABLE_APPROX(test_cart_coords.get(j), cart_coords.get(j));
+  }
+
+  // Test surface_normal_one_form
+  StrahlkorperDB::types<Frame::Inertial>::OneForm test_normal_one_form(n_pts);
+  {
+    const auto& r = db::get<StrahlkorperDB::Radius<Frame::Inertial>>(box);
+    const DataVector temp = r + amp * sin_phi * sin_theta;
+    const DataVector one_over_r = 1.0 / r;
+    test_normal_one_form.get(0) = cos_phi * sin_theta * temp * one_over_r;
+    test_normal_one_form.get(1) =
+        (sin_phi * sin_theta * temp - amp) * one_over_r;
+    test_normal_one_form.get(2) = cos_theta * temp * one_over_r;
+  }
+  const auto& normal_one_form =
+      db::get<StrahlkorperDB::NormalOneForm<Frame::Inertial>>(box);
+  for (size_t j = 0; j < 3; ++j) {
+    CHECK_ITERABLE_APPROX(test_normal_one_form.get(j), normal_one_form.get(j));
+  }
+
+  // Test surface_normal_magnitude.
+  tnsr::II<DataVector, 3, Frame::Inertial> invg(n_pts);
+  invg.get(0, 0) = 1.0;
+  invg.get(1, 0) = 0.1;
+  invg.get(2, 0) = 0.2;
+  invg.get(1, 1) = 2.0;
+  invg.get(1, 2) = 0.3;
+  invg.get(2, 2) = 3.0;
+
+  DataVector test_normal_mag(n_pts);
+  {
```

Comment:
I would prefer a lambda setting test_normal_mag since then it's clear that this just an initialization of that.